### PR TITLE
Create machines for bluegreen deploy concurrently

### PR DESCRIPTION
**Pull request by tvdfly**
Previously, we created the green (new) machines sequentially when running a bluegreen deployment. That works ok when there are a few machines, and it becomes a bit slow when there are 70 or 100 or 500+ machines in the app.

Update the CreateGreenMachines() func to use the pool pattern we have in a few places.

I did the same thing in the MarkGreenMachinesAsReadyForTraffic(), CordonBlueMachines(), StopBlueMachines() funcs.

---

### Documentation

- [ ] Fresh Produce
- [ ] In superfly/docs, or asked for help from docs team
- [ ] n/a

### CODE REVIEW

**Comment by tvdfly** [comment]
The `TestConfigWatch` test is failing on macos. It doesn't seem related to this change and I wasn't able to repro. I think we can take care of it in a separate pr.

File: internal/command/deploy/machines.go [hunk 1]
```diff
@@ -191,6 +193,11 @@ func NewMachineDeployment(ctx context.Context, args MachineDeploymentArgs) (Mach
 		maxUnavailable = *appConfig.Deploy.MaxUnavailable
 	}
 
+	bluegreenMaxConcurrent := args.BlueGreenMaxConcurrent
+	if bluegreenMaxConcurrent < 1 {
+		bluegreenMaxConcurrent = 1
+	}
+
```
Comment (benbjohnson) [inline internal/command/deploy/machines.go:200]: Can we consolidate this down to a single `--max-concurrent` flag? Or do we need a separate bluegreen & immediate flag? For backwards compatibility, we could have `--immediate-max-concurrent` still work but maybe just hide it from the docs?

**Review by benbjohnson — APPROVED** [review]
Overall lgtm. Just a question about the flag name.

File: internal/command/deploy/machines.go (hunk 1 above)
Comment (tvdfly) [reply internal/command/deploy/machines.go:200]: i like that, i'll update

**Pull request merged**